src/change.c

```diff
@@ -1194,14 +1197,32 @@ ins_str(char_u *s)
     oldp = ml_get(lnum);
     oldlen = (int)ml_get_len(lnum);
 
-    newp = alloc(oldlen + newlen + 1);
-    if (newp == NULL)
-	return;
-    if (col > 0)
+    newtotlen = oldlen + newlen;
+    new_size = newtotlen + 1;
+    if ((curbuf->b_ml.ml_line_size < new_size) || (prevp != oldp)
+#ifdef FEAT_PROP_POPUP
+	    || curbuf->b_has_textprop
+#endif
+	    )
+    {
+#ifdef FEAT_PROP_POPUP
+	if (!curbuf->b_has_textprop)
+#endif
+	    new_size = new_size * 3 / 2;
+	newp = alloc(new_size);
+	if (newp == NULL)
+	    return;
+	curbuf->b_ml.ml_line_size = new_size;
```

Comment:
That should potentially be encapsulated into a new memline()  function `ml_replace_len_size()` which takes the new size parameter and stores the new size there.

Reply:
I tried to implement this, but it didn't work...